lib/python/dataset.cpp

```diff
@@ -107,14 +108,14 @@ template <class T> void bind_rebin(py::module &m) {
       py::call_guard<py::gil_scoped_release>());
 }
 
-template <class Value> auto to_cpp_dim_map(const py::dict &dict) {
-  std::unordered_map<Dim, Value> out;
+template <class Key, class Value> auto to_cpp_map(const py::dict &dict) {
```

Comment:
scipp is very sparse in terms of cpp function docstrings/doxygen, but here it may be worth adding something that warns about using this without GIL release.

Reply:
Isn't this clear from the argument type?

Reply:
Probably.